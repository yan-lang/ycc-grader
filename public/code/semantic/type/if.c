int main() {
    int a = 10;
    int b = 9;
    if(a+b>0) {
        a = b;
    } else {
        b = a;
    }

    if(a > b) {
        a = b;
    }
    return 0;
}