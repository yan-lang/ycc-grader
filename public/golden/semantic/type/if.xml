<TranslationUnit>
    <FuncDecl>
        <name>
            <Id>main</Id>
        </name>
        <returnType>
            <PrimitiveType>INT</PrimitiveType>
        </returnType>
        <params>
        </params>
        <body>
            <Block>
                <VarDecl>
                    <name>
                        <Id>a</Id>
                    </name>
                    <type>
                        <PrimitiveType>INT</PrimitiveType>
                    </type>
                    <init>
                        <Literal evalType="int" type="INT">10</Literal>
                    </init>
                </VarDecl>
                <VarDecl>
                    <name>
                        <Id>b</Id>
                    </name>
                    <type>
                        <PrimitiveType>INT</PrimitiveType>
                    </type>
                    <init>
                        <Literal evalType="int" type="INT">9</Literal>
                    </init>
                </VarDecl>
                <IfStmt>
                    <cond>
                        <BinaryExpr evalType="bool" op="GT">
                            <lhs>
                                <BinaryExpr evalType="int" op="ADD">
                                    <lhs>
                                        <Id evalType="int">a</Id>
                                    </lhs>
                                    <rhs>
                                        <Id evalType="int">b</Id>
                                    </rhs>
                                </BinaryExpr>
                            </lhs>
                            <rhs>
                                <Literal evalType="int" type="INT">0</Literal>
                            </rhs>
                        </BinaryExpr>
                    </cond>
                    <thenBody>
                        <Block>
                            <ExprStmt>
                                <AssignExpr evalType="int">
                                    <assignee>
                                        <Id evalType="int">a</Id>
                                    </assignee>
                                    <value>
                                        <Id evalType="int">b</Id>
                                    </value>
                                </AssignExpr>
                            </ExprStmt>
                        </Block>
                    </thenBody>
                    <elseBody>
                        <Block>
                            <ExprStmt>
                                <AssignExpr evalType="int">
                                    <assignee>
                                        <Id evalType="int">b</Id>
                                    </assignee>
                                    <value>
                                        <Id evalType="int">a</Id>
                                    </value>
                                </AssignExpr>
                            </ExprStmt>
                        </Block>
                    </elseBody>
                </IfStmt>
                <IfStmt>
                    <cond>
                        <BinaryExpr evalType="bool" op="GT">
                            <lhs>
                                <Id evalType="int">a</Id>
                            </lhs>
                            <rhs>
                                <Id evalType="int">b</Id>
                            </rhs>
                        </BinaryExpr>
                    </cond>
                    <thenBody>
                        <Block>
                            <ExprStmt>
                                <AssignExpr evalType="int">
                                    <assignee>
                                        <Id evalType="int">a</Id>
                                    </assignee>
                                    <value>
                                        <Id evalType="int">b</Id>
                                    </value>
                                </AssignExpr>
                            </ExprStmt>
                        </Block>
                    </thenBody>
                </IfStmt>
                <ReturnStmt>
                    <value>
                        <Literal evalType="int" type="INT">0</Literal>
                    </value>
                </ReturnStmt>
            </Block>
        </body>
    </FuncDecl>
</TranslationUnit>